test-runner-poc/environments.js

```diff
@@ -0,0 +1,267 @@
+'use strict';
+
+const f = require('util').format;
+const semver = require('semver');
+const path = require('path');
+const core = require('../lib/core');
+
+/**
+ * Base class for environments in projects that use the test
+ * runner
+ */
+class EnvironmentBase {
+  /**
+   * The default implementation of the environment setup
+   *
+   * @param {*} callback
+   */
+  setup(callback) {
+    callback();
+  }
+}
+
+const genReplsetConfig = (port, options) => {
+  return Object.assign(
+    {
+      options: {
+        bind_ip: 'localhost',
+        port: port,
+        dbpath: `${__dirname}/../db/${port}`,
+        setParameter: ['enableTestCommands=1']
+      }
+    },
+    options
+  );
+};
+
+function usingUnifiedTopology() {
+  return !!process.env.MONGODB_UNIFIED_TOPOLOGY;
+}
+
+/**
+ *
+ * @param {*} discoverResult
+ */
+class ReplicaSetEnvironment extends EnvironmentBase {
+  constructor(host, port, version) {
+    super();
+
+    this.host = host;
+    this.port = port;
+    this.setName = 'replset';
+    this.url = `mongodb://%s${host}:${port}/integration_tests?rs_name=rs`;
```

Comment:
The `url` is also dynamically created with the passed in `host` and `port`. This change is also reflected in `ReplicaSetEnvironment`, `ShardedEnvironment`, `SslEnvironment` and `AuthEnvironment`.